### PR TITLE
func_curl: add new headers or replace them if already exists

### PR DESCRIPTION
**Pull request by scgm11**
Resolves: #135

### CODE REVIEW

**Comment by scgm11** [comment]
cherry-pick-to: 20
cherry-pick-to: 18

File: funcs/func_curl.c [hunk 1]
```diff
@@ -332,6 +332,37 @@ static int parse_curlopt_key(const char *name, CURLoption *key, enum optiontype
...
+                } else {
+                	ast_log(LOG_ERROR, "Unable to allocate memory for new header: %s\n", header);
+                    return;
+                }
+            }
+        }
+        prev = current;
+        current = current->next;
+    }
+
```
Comment (scgm11) [inline funcs/func_curl.c:362]: I think I may be missing a curl_slist_free_all(current); here ?

File: funcs/func_curl.c [hunk 2]
```diff
@@ -698,7 +729,7 @@ static int acf_curl_helper(struct ast_channel *chan, struct curl_args *args)
 		if (cur->key == CURLOPT_SPECIAL_HASHCOMPAT) {
 			hashcompat = (long) cur->value;
 		} else if (cur->key == CURLOPT_HTTPHEADER) {
-			headers = curl_slist_append(headers, (char*) cur->value);
+			replace_or_add_header(&headers, (char*) cur->value);
```
Comment (jcolp) [inline funcs/func_curl.c:732]: I do not believe it should be the responsibility of the acf_curl_helper function to do this replacement. The list of headers on the datastore should only ever contain a single header of the given name. The global headers list should not add a header to the request if the same header is present in the datastore list.

**Review by jcolp — CHANGES_REQUESTED** [review]

File: funcs/func_curl.c (hunk 2 above)
Comment (scgm11) [reply funcs/func_curl.c:732]: so you want the change in acf_curlopt_write ath the end where 	AST_LIST_INSERT_TAIL(list, new, list); ?

Comment (jcolp) [reply funcs/func_curl.c:732]: That is where I would expect such functionality to be written, but there would also have to be logic in acf_curl_helper for the global / channel datastore situation.

Comment (scgm11) [reply funcs/func_curl.c:732]: ok sorry cant understand what you want, you can close this if you feel its not good enough as I don't have the knowledge to do it as you want.

**Comment by jcolp** [comment]
Closed per comments, as well as comment on issue regarding implementation.

**Pull request closed without merging**